contracts/api3-server-v1/proxies/ProxyFactoryZkSync.sol

```diff
@@ -0,0 +1,243 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./DataFeedProxy.sol";
+import "./DapiProxy.sol";
+import "./DataFeedProxyWithOev.sol";
+import "./DapiProxyWithOev.sol";
+import "./interfaces/IProxyFactory.sol";
+
+/// @title Contract factory that deterministically deploys proxies that read
+/// data feeds (Beacons or Beacon sets) or dAPIs, along with optional OEV
+/// support
+/// @notice This ProxyFactory implementation replaces the OpenZeppelin CREATE2
+/// address computation to match zkSync's. Accordingly, only use this contract
+/// on the zkSync chains that use the ContractDeployer contract to deploy
+/// contracts. For more information, see the link below
+/// https://github.com/matter-labs/era-system-contracts/blob/857da713d9756ec13f071916848e283e74579817/contracts/ContractDeployer.sol
+/// @dev The proxies are deployed normally and not cloned to minimize the gas
+/// cost overhead while using them to read data feed values
+contract ProxyFactoryZkSync is IProxyFactory {
+    // https://github.com/matter-labs/era-system-contracts/blob/857da713d9756ec13f071916848e283e74579817/contracts/Constants.sol#L74
+    bytes32 private constant CREATE2_PREFIX =
+        0x2020dba91b30cc0006188af794c2fb30dd8520db7e2c088b7fc7c103c00ca494;
+    // According to the zkSync docs, bytecode is hashed using sha256, and then the first two bytes are
+    // replaced with the length of the bytecode in 32-byte words.
+    // To verify the hashes below
+    // - Build https://github.com/api3dao/airnode-protocol-v1/tree/deploy-zksync-reference
```

Comment:
I'm planning to leave ProxyFactoryZkSync.sol, its tests and the zkSync dependencies on https://github.com/api3dao/airnode-protocol-v1/tree/deploy-zksync-reference and only transfer over the deployment files (as it's currently done on main). I just added a branch protection rule to the deploy-zksync-reference to prevent it from being deleted.

A long term solution could be a separate zksync repo that imports @api3/airnode-protocol-v1, implements and tests ProxyFactoryZkSync.sol and deploys the contracts, but the current solution works fine so I don't see the advantage.